src/recordlinker/linking/matchers.py

```diff
@@ -18,10 +19,54 @@
 
 SIMILARITY_MEASURES = typing.Literal["JaroWinkler", "Levenshtein", "DamerauLevenshtein"]
 
-# A Callable type for comparing a feature on two records
-FEATURE_COMPARE_FUNC = typing.Callable[[PIIRecord, Patient, Feature, typing.Any], float]
-# A Callable type for evaluating whether a set of feature comparisons constitutes a match
-MATCH_RULE_FUNC = typing.Callable[[list[float], typing.Any], bool]
+
+class RuleFunc(enum.Enum):
+    """
+    Enum for the different types of match rules that can be used for patient
+    matching. This is the universe of all possible match rules that a user can
+    choose from when configuring their algorithm.  When data is loaded into the
```

Comment:
I like that we constrain the match rules that the user can select based on those in the default Basic and Enhanced algos (i.e., that showed highest match accuracy based on performance testing and lit review)